### ISSUE TITLE
Feature request: a red notification bubble on the app icon when there is an outstanding survey past due

### ISSUE THREAD

**Issue by tlakhtak** [BIDMCDigitalPsychiatry/LAMP-platform#179]
> Feature request: a red notification bubble on the app icon when there is an outstanding survey past due

**Comment by avaidyam** [comment]
On Android devices, the bubble will appear whenever there are notifications from the app that the user hasn't dismissed yet. 

@jijopulikkottil Is this the same behavior on iOS as well? (i.e. If there are outstanding notifications that were not tapped/swiped away, does the app show the red bubble?)

**Comment by jijopulikkottil** [comment]
@avaidyam 
>  Is this the same behavior on iOS as well?
No. iOS app doesn't show any badge.

**Comment by avaidyam** [comment]
Can you enable it to show a badge if there are notifications? [It looks like this is a property that can be sent from the server APNS payload.](https://developer.apple.com/documentation/usernotifications/unmutablenotificationcontent/1649875-badge)

**Comment by jijopulikkottil** [comment]
@avaidyam In iOS, It is an integer value. But it might be difficult to maintain the pending unread notifications (or unattended activities? @Linoy339) count on server side. Let us know if we need to implement it. 
Or should we set badge = 1 for any notification from server side ?, and clear it when opening the app ?.

**Comment by avaidyam** [comment]
I see what you mean, because this would require the server to keep track of the state of how many activities have not been opened by the patient, and then send a silent push notification to erase the badge.  This would however be zero effort on the iOS / Swift codebase side, right? Furthermore, since the Android app already has this behavior by default, I think it makes sense to have the state managed by the server-side. @Linoy339

**Comment by avaidyam** [comment]
@jijopulikkottil [Could you evaluate whether this method allows the badge to auto increment on the client app side?](https://www.cometchat.com/docs/)

**Comment by jijopulikkottil** [comment]
> server to keep track of the state of how many activities have not been opened by the patient, and then send a silent push notification to erase the badge. This would however be zero effort on the iOS / Swift codebase side, right? 

Yes


@avaidyam  the above link navigate to comet chat docs. 

> Furthermore, since the Android app already has this behavior by default

Not sure about the Android behaviour, I think In android app it shows red dot until the app open.. right?

Do we need to try to increment the badge manually when receive remote notification and reset it when open the app ?

**Comment by avaidyam** [comment]
@jijopulikkottil Apologies about the dead link! Let's stick to a client-side implementation on iOS only. The badge number can be autoincremented in a notification service extension and decremented when a notification is swiped away. [Here's the correct link.](https://www.cometchat.com/docs/ios-chat-sdk/increment-app-icon-badge-count)

**Comment by jijopulikkottil** [comment]
Ok, our plan is to increment the badge manually when receive remote notification.
@avaidyam  And instead of resetting to 0 when open the app, we have to decrement the badge count  when user tap/swipe on it or when expired.. right ?

**Comment by avaidyam** [comment]
Yes - the badge number should reflect the number of notifications that are still pending. Are the notifications cleared when the app is opened right now? If so, the behavior you suggest (the badge count is reset when the app is opened) will happen automatically.

**Comment by avaidyam** [comment]
Great, that sounds like a very good implementation! Thanks.

**Comment by ZCOEngineer** [comment]
Nothing to do on backend

**Comment by jijopulikkottil** [comment]
Yes, It is handled on app side. And the updated version 2022.1.13 is in TestFlight now.

**Comment by avaidyam** [comment]
So this is already being QA tested? If so, please add the GitHub label.

**Comment by jijopulikkottil** [comment]
Yes, It is QA tested.

**Comment by avaidyam** [comment]
This build has now been submitted to App Store review, after which we'll QA test it and push for production release.
